### PR TITLE
Add drag and drop callback

### PR DESCRIPTION
**Pull request by alexanderbock**

### CODE REVIEW

File: scripts/drag_drop_handler.lua [hunk 1]
```diff
@@ -0,0 +1,37 @@
+-- this script gets two parameters in its global scope:
```
Comment (WeirdRubberDuck) [inline scripts/drag_drop_handler.lua:1]: Nitpicky, but this sentence should start with a capital letter

**Review by WeirdRubberDuck — APPROVED** [review]
😍

File: scripts/drag_drop_handler.lua (hunk 1 above)
Comment (alexanderbock) [reply scripts/drag_drop_handler.lua]: not need for "but";  "nitpicky" and "right" are not opposites

File: scripts/drag_drop_handler.lua [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  do return "" end
+end
+
+-- Lua doesn't enjoy \ that are used by Windows extensively. So we convert all \ into /
+filename = filename:gsub("\\", "/")
+basename = basename:gsub("\\", "/")
+basename_without_extension = basename:sub(0, #basename - extension:len())
+
+if     extension == ".jpg" or extension == ".jpeg" or extension == ".tif" or
+       extension == ".png" then
```
Comment (GPayne) [inline scripts/drag_drop_handler.lua:26]: I could see this section being added-to in the future with new file types. How about adding functions to check for file types, like:
`if isImage() then`
The function itself would test for the 4 file image types (a list which may expand) and make the evaluation of file types cleaner.

**Review by GPayne — APPROVED** [review]

File: scripts/drag_drop_handler.lua (hunk 2 above)
Comment (alexanderbock) [reply scripts/drag_drop_handler.lua]: Good idea!

File: scripts/drag_drop_handler.lua [hunk 3]
```diff
@@ -0,0 +1,43 @@
+-- This script gets two parameters in its global scope:
+-- filename:   The full path for the file that was dropped on the application.
+--             Example:  C:/OpenSpace/openspace.cfg
+-- basename:   Only the name of the actual file with extension, but without the full rest
+--             of the path.
+--             Example:  openspace.cfg
+-- extension:  The extention of the file
+--             Example: cfg
```
Comment (Roxeena) [inline scripts/drag_drop_handler.lua:8]: Shouldn't this be .cfg instead of cfg?

**Review by Roxeena — APPROVED** [review]

File: scripts/drag_drop_handler.lua (hunk 3 above)
Comment (alexanderbock) [reply scripts/drag_drop_handler.lua]: Good eye!  It definitely should be

File: src/engine/openspaceengine.cpp [hunk 4]
```diff
@@ -1444,6 +1445,52 @@ void OpenSpaceEngine::touchExitCallback(TouchInput input) {
     }
 }
 
+void OpenSpaceEngine::handleDragDrop(const std::string& file) {
+    std::filesystem::path f(file);
+
+    ghoul::lua::LuaState s(ghoul::lua::LuaState::IncludeStandardLibrary::Yes);
+    std::string absolutePath = absPath("${BASE}/scripts/drag_drop_handler.lua");
```
Comment (micahnyc) [inline src/engine/openspaceengine.cpp:1452]: ${SCRIPTS} vs ${BASE}/scripts

Comment (alexanderbock) [reply src/engine/openspaceengine.cpp]: Good point. Fixed

**Pull request merged**